pkg/operator/osd/osd.go

```diff
@@ -71,6 +80,12 @@ func New(context *clusterd.Context, namespace, version string, storageSpec Stora
 func (c *Cluster) Start() error {
 	logger.Infof("start running osds in namespace %s", c.Namespace)
 
+	// create the artifacts for the api service to work with RBAC enabled
+	err := k8sutil.MakeClusterRole(c.context.Clientset, c.Namespace, appName, clusterAccessRules)
```

Comment:
The operator role is cross namespaces right?  I only want an OSD to be able to perform operations within one namespace and I want its permissions to be scoped to the minimal set it needs to do its job, which is manipulate config maps.  I don't want to give an OSD the full set of permissions that the operator gets.

Reply:
Ah.. I see it a bit differently. I see Rook as a single application/service that users deploy and that is composed of many subcomponents like operator, agents, mon, osd, mds, etc. And that is why i was advocating on having just one RBAC role that gets reused on all rook components instead of having each subcomponents define their own. If each subcomponent was usable and can stand on its own then i guess it would make more sense to have separate RBAC roles for each.